File: bot.py
import discord
import json

token = ''
with open('settings.json', 'r') as f:
    token = json.load(f)["bot_token"]

client = discord.Client()

@client.event
async def on_ready():
    print(f'[*] {client.user} has connected to Discord!')

client.run(token)

File: src/bot.ts
import path from 'path'
import { Client } from 'discord.js'
import { readJSONSync } from 'fs-extra'

const PATH = path.resolve()
const { bot_token : token } = readJSONSync(PATH + '/settings.json')
export const client = new Client()

client.on('ready', () => console.log('[*] Ready'))

client.login(token)